### ISSUE TITLE
How would a forwardf work?

### ISSUE THREAD

**Issue by TWith2Sugars** [SaturnFramework/Saturn#25]
I'm attempting to create a forwardf extension to scope so that I can pass along an id to a sub resource.

```
let subResource projectId = scope {
  get "/" getAllItems projectId
  getf "/%i" getItem projectId
}

let project = scope {
  get "/" allProjects
  getf "/%i" getProject
  forwardf "%i/items" subResource 
}
```

that snippet probably explains it best on what I'm trying do.

I'm happy to create a pr and work on this but I'm not 100% sure how to go about this.

**Comment by Krzysztof-Cieslak** [comment]
TBF, I'm not entirely sure why it's not implemented... Probably I've hit some problems with type system (scope implementation is bit hacky). But anyway, from top of my head, it should be fairly similar to other format function implementations in `Router.fs`  - using `addRouteF` helper function.

**Comment by Krzysztof-Cieslak** [comment]
I'm not sure if I 100% agree with @gerardtoconnor about this :D

**Comment by gerardtoconnor** [comment]
Im no phoenix / Saturn expert so dont mind me!

**Comment by Krzysztof-Cieslak** [comment]
Oh, that has nothing really to do Phoenix nor Saturn... I think

**Comment by gerardtoconnor** [comment]
My statement still stands, don't mind me ;)

**Comment by Krzysztof-Cieslak** [comment]
Anyway, I believe that I've accidentally implemented `fowardf` adding some stuff to `controller` today - https://github.com/SaturnFramework/Saturn/blob/master/src/Saturn/Controller.fs#L68

**Comment by TWith2Sugars** [comment]
well...., I think I'll still give it a shot at the scope level anyway :) .

**Comment by TWith2Sugars** [comment]
Well looks like [subroutef](https://github.com/giraffe-fsharp/Giraffe/commit/224f4a3ee91bb6ff3c97d06e0d4dbe342025edd1) now exists, I'll wait until it's released and base my attempt around that.

**Comment by Krzysztof-Cieslak** [comment]
Implemented in 756a3abc6f545844ea13c9b060903e5d654b05ff

**Comment by TWith2Sugars** [comment]
Thanks for doing this, I was getting no where with it.
